### PR TITLE
feat: allow users to specify an alternative init directory

### PR DESCRIPTION
**Pull request by josegonzalez**
In some existing environments, packages might already use another environment for the purpose this `/etc/my_init.d` directory serves.

### CODE REVIEW

File: image/bin/my_init [hunk 1]
```diff
@@ -249,9 +251,9 @@ def kill_all_processes(time_limit):
 
 
 def run_startup_files():
-    # Run /etc/my_init.d/*
-    for name in listdir("/etc/my_init.d"):
-        filename = "/etc/my_init.d/" + name
+    # Run ENV_INIT_DIRECTORY*
```
Comment (hyperknot) [inline image/bin/my_init]: I'd write `ENV_INIT_DIRECTORY/*` but that's just a nitpicking.

**Comment by hyperknot** [comment]
Looks good as well!

File: image/bin/my_init (hunk 1 above)
Comment (Theaxiom) [inline image/bin/my_init]: @hyperknot fixed

**Comment by hyperknot** [comment]
@josegonzalez want to merge this?

**Comment by josegonzalez** [comment]
This isn't my repository, this is just a pull request.

**Comment by hyperknot** [comment]
OK, so is there still a need for this? If yes I'll merge it.

**Comment by hyperknot** [comment]
On the other hand the string `my_init.d` is present in a lot of files (like when created with mkdir -p), and I'd personally prefer not to change it. What is the use case for customising it?

**Comment by josegonzalez** [comment]
We have an existing system that writes env files to `/etc/default`.

**Comment by kingdonb** [comment]
So, user went somewhere else and we have a fix passing CI that might have kept them around...

**Comment by kingdonb** [comment]
Maybe I've overblown the impact of these commits by a scosche... it looks like these commits just parameterize something that wasn't a parameter before.  I vote to merge

**Comment by kingdonb** [comment]
I put these commits into #432 rebased onto master

**Comment by kingdonb** [comment]
I ran the CI checks on #432 and they passed, @Theaxiom @hyperknot if either of you review there then I can approve and merge.

(Closing)

**Pull request closed without merging**